### PR TITLE
Default to only show language detection hints in text editors

### PR DESCRIPTION
**Pull request by JacksonKearl**
Fixes https://github.com/microsoft/vscode/issues/148220

### CODE REVIEW

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Con
 			},
 			'workbench.editor.languageDetectionHints': {
 				type: 'string',
-				default: 'always',
+				default: 'textEditors',
 				tags: ['experimental'],
 				enum: ['always', 'notebookEditors', 'textEditors', 'never'],
```
Comment (roblourens) [inline src/vs/workbench/browser/workbench.contribution.ts:118]: An object `{ notebookEditors: true, textEditors: true }` would be more extensible to more than 2 editor types, if that would ever be relevant

**Review by roblourens — APPROVED** [review]

**Review by roblourens — APPROVED** [review]

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 2]
```diff
@@ -112,17 +112,21 @@ const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Con
 				description: localize('workbench.editor.preferBasedLanguageDetection', "When enabled, a language detection model that takes into account editor history will be given higher precedence."),
 			},
 			'workbench.editor.languageDetectionHints': {
-				type: 'string',
-				default: 'always',
+				type: 'object',
+				default: { 'untitledEditors': true, 'notebookEditors': false },
```
Comment (roblourens) [inline src/vs/workbench/browser/workbench.contribution.ts:116]: Maybe `untitledTextEditors`?

Comment (roblourens) [reply src/vs/workbench/browser/workbench.contribution.ts:116]: Untitled notebook editors exist too

Comment (JacksonKearl) [reply src/vs/workbench/browser/workbench.contribution.ts:116]: Clarified in description 👍

**Pull request merged**